From 74c048cb5a80b3f44fa0d38e71003b0d119c8d93 Mon Sep 17 00:00:00 2001
From: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
Date: Wed, 15 Aug 2018 17:53:31 +0700
Subject: [PATCH 3/5] qtmultimedia: replace playbin by playbin3 to play video

select playbin3 instead of playbin to play some special videos
remove some unnecessary properties on playbin3 to avoild warning

Signed-off-by: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
---
 src/multimedia/gsttools_headers/qgstutils_p.h        |  2 +-
 .../mediaplayer/qgstreamerplayersession.cpp          | 12 ++++++------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/multimedia/gsttools_headers/qgstutils_p.h b/src/multimedia/gsttools_headers/qgstutils_p.h
index 124805727..d27792fc9 100644
--- a/src/multimedia/gsttools_headers/qgstutils_p.h
+++ b/src/multimedia/gsttools_headers/qgstutils_p.h
@@ -57,7 +57,7 @@
 #include <QDebug>
 
 #if GST_CHECK_VERSION(1,0,0)
-# define QT_GSTREAMER_PLAYBIN_ELEMENT_NAME "playbin"
+# define QT_GSTREAMER_PLAYBIN_ELEMENT_NAME "playbin3"
 # define QT_GSTREAMER_CAMERABIN_ELEMENT_NAME "camerabin"
 # define QT_GSTREAMER_COLORCONVERSION_ELEMENT_NAME "videoconvert"
 # define QT_GSTREAMER_RAW_AUDIO_MIME "audio/x-raw"
diff --git a/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp b/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
index a1eee0166..3f8b41d3b 100644
--- a/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
+++ b/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
@@ -238,9 +238,9 @@ QGstreamerPlayerSession::QGstreamerPlayerSession(QObject *parent)
             g_signal_connect(G_OBJECT(m_playbin), "notify::mute", G_CALLBACK(handleMutedChange), this);
         }
 
-        g_signal_connect(G_OBJECT(m_playbin), "video-changed", G_CALLBACK(handleStreamsChange), this);
-        g_signal_connect(G_OBJECT(m_playbin), "audio-changed", G_CALLBACK(handleStreamsChange), this);
-        g_signal_connect(G_OBJECT(m_playbin), "text-changed", G_CALLBACK(handleStreamsChange), this);
+//        g_signal_connect(G_OBJECT(m_playbin), "video-changed", G_CALLBACK(handleStreamsChange), this);
+//        g_signal_connect(G_OBJECT(m_playbin), "audio-changed", G_CALLBACK(handleStreamsChange), this);
+//        g_signal_connect(G_OBJECT(m_playbin), "text-changed", G_CALLBACK(handleStreamsChange), this);
 
 #if defined(HAVE_GST_APPSRC)
         g_signal_connect(G_OBJECT(m_playbin), "deep-notify::source", G_CALLBACK(configureAppSrcElement), this);
@@ -1350,9 +1350,9 @@ void QGstreamerPlayerSession::getStreamsInfo()
     gint videoStreamsCount = 0;
     gint textStreamsCount = 0;
 
-    g_object_get(G_OBJECT(m_playbin), "n-audio", &audioStreamsCount, NULL);
-    g_object_get(G_OBJECT(m_playbin), "n-video", &videoStreamsCount, NULL);
-    g_object_get(G_OBJECT(m_playbin), "n-text", &textStreamsCount, NULL);
+//    g_object_get(G_OBJECT(m_playbin), "n-audio", &audioStreamsCount, NULL);
+//    g_object_get(G_OBJECT(m_playbin), "n-video", &videoStreamsCount, NULL);
+//    g_object_get(G_OBJECT(m_playbin), "n-text", &textStreamsCount, NULL);
 
     haveAudio = audioStreamsCount > 0;
     haveVideo = videoStreamsCount > 0;
-- 
2.17.1

